src/App/Pages/Accounts/LoginSsoPageViewModel.cs

```diff
@@ -209,17 +209,12 @@ private async Task LogIn(string code, string codeVerifier, string orgId)
                 if (response.TwoFactor)
                 {
                     StartTwoFactorAction?.Invoke();
+                    return;
                 }
-                else if (response.ResetMasterPassword)
-                {
-                    StartSetPasswordAction?.Invoke();
-                }
-                else if (response.ForcePasswordReset)
-                {
-                    UpdateTempPasswordAction?.Invoke();
-                }
```

Comment:
Do we need to leave these to support backwards compatibility? @JaredSnider-Bitwarden did we do that in the clients?

Reply:
Yes, in the existing flow, those can both still happen and should remain.  

Note: we replaced the usage of `ResetMasterPassword` with:
```TypeScript
const requireSetPassword =
        !acctDecryptionOpts.hasMasterPassword &&
        acctDecryptionOpts.keyConnectorOption === undefined;
 ```
 
 For the TDE routing path, we have to handle 3 routing scenarios (code taken from `sso.component.ts` in clients):
 ```TypeScript
   private async handleTrustedDeviceEncryptionEnabled(
    authResult: AuthResult,
    orgIdentifier: string,
    acctDecryptionOpts: AccountDecryptionOptions
  ): Promise<void> {
    // If user doesn't have a MP, but has reset password permission, they must set a MP
    if (
      !acctDecryptionOpts.hasMasterPassword &&
      acctDecryptionOpts.trustedDeviceOption.hasManageResetPasswordPermission
    ) {
      // Change implies going no password -> password in this case
      return await this.handleChangePasswordRequired(orgIdentifier);
    }

    if (authResult.forcePasswordReset !== ForceResetPasswordReason.None) {
      return await this.handleForcePasswordReset(orgIdentifier);
    }

    this.navigateViaCallbackOrRoute(
      this.onSuccessfulLoginTdeNavigate,
      // Navigate to TDE page (if user was on trusted device and TDE has decrypted
      //  their user key, the login-initiated guard will redirect them to the vault)
      [this.trustedDeviceEncRoute]
    );
  }
 ```

Reply:
just need to update the condition because i was looking for the KeyConnectorOption instead of the hasManageResetPasswordPermission

Reply:
oh, great catch. Didn't realize that we needed to support that for new clients. Wasn't thinking about new clients + older servers. I'll fix that on clients